tests/srv6/test_srv6.py

```diff
@@ -0,0 +1,170 @@
+from tests.common.helpers.assertions import pytest_assert as py_assert
+
+pytestmark = [
+    pytest.mark.topology('t0')
+]
+
+Logger = logging.getLogger(__name__)
+
+V6_PREFIX_NBR = "2001:db8::"
+V6_MASK_NBR = "64"
+
+V6_PREFIX_DUT = "2001:db8:1::"
+V6_MASK_DUT = "64"
+
+
+@pytest.fixture(name="setUp", scope="module")
+def fixture_setUp(nbrhosts, duthosts, enum_frontend_dut_hostname):
+
+    # pick dut
+    duthost = duthosts[enum_frontend_dut_hostname]
+
+    # pick one neighbor
+    nbr = None
+    nbrnames = list(nbrhosts.keys())
+    for name in nbrnames:
+        if 'T1' in name:
+            nbr = nbrhosts[name]
+            break
+    py_assert(nbr is not None, "No T1 neighbors")
+
+    yield (nbr, duthost)
+
+    Logger.info("Performing cleanup")
+
+    # cleanup neighbor
+    cmd = "config vrf del Vrf10"
+    nbr['host'].shell(cmd)
+
+    # cleanup dut
+    duthost = duthosts[enum_frontend_dut_hostname]
+    cmd = "config vrf del Vrf10"
+    duthost.shell(cmd)
+
+
+def run_srv6(enum_frontend_dut_hostname, hosts):
+    """ Route added on All neighbor should be learned by the DUT"""
+    Logger.info("Adding routes on neighbors")
+
+    nbr = hosts[0]
+    duthost = hosts[1]
+
+    # configure bgp on neighbor
+    cmd = "config vrf add Vrf10"
+    nbr['host'].shell(cmd)
+    cmd = "config loopback add Loopback1"
+    nbr['host'].shell(cmd)
+    cmd = "config interface vrf bind Loopback1 Vrf10"
+    nbr['host'].shell(cmd)
+    cmd = "config interface ip add Loopback1 2001:db8:2::1"
+    nbr['host'].shell(cmd)
+    cmd = ("vtysh"
+           " -c 'configure'"
+           " -c 'ipv6 route {}/{} Loopback1'"
+           " -c 'segment-routing'"
+           " -c 'srv6'"
+           " -c 'locators'"
+           " -c 'locator MAIN'"
+           " -c 'prefix fcbb:bbbb:1::/48 block-len 32 node-len 16 func-bits 16'"
+           " -c 'exit'"
+           " -c 'exit'"
+           " -c 'exit'"
+           " -c 'exit'"
+           " -c 'router bgp 64600'"
+           " -c 'address-family ipv6 unicast'"
+           " -c 'network {}/{}'"
+           " -c 'exit'"
+           " -c 'segment-routing srv6'"
+           " -c 'locator MAIN'"
+           " -c 'exit'"
+           " -c 'router bgp 64600 vrf Vrf10'"
+           " -c 'address-family ipv6 unicast'"
+           " -c 'sid vpn export auto'"
+           " -c 'rd vpn export 1:10'"
+           " -c 'rt vpn both 99:99'"
+           " -c 'import vpn'"
+           " -c 'export vpn'"
+           " -c 'redistribute connected'"
+           " -c 'exit'"
+           " -c 'exit'").format(V6_PREFIX_NBR, V6_MASK_NBR, V6_PREFIX_NBR, V6_MASK_NBR)
+    nbr['host'].shell(cmd)
+    Logger.info("Route %s added", V6_PREFIX_NBR)
+
+    # configure bgp on dut
+    cmd = "config vrf add Vrf10"
+    duthost.shell(cmd)
+    cmd = "config loopback add Loopback1"
+    duthost.shell(cmd)
+    cmd = "config interface vrf bind Loopback1 Vrf10"
+    duthost.shell(cmd)
+    cmd = "config interface ip add Loopback1 2001:db8:2::1"
```

Comment:
although this may not have any functional impact, but we'd better use a different Loopback address for DUT instead of reusing the Loopback address.